### PR TITLE
Move azure client under service-clients directory 

### PR DESCRIPTION
**Pull request by sonalideshpandemsft**
This PR moves azure-client under service-clients directory. The motivation behind this adjustment is primarily for organizational purposes.

[AB#6624](https://dev.azure.com/fluidframework/internal/_workitems/edit/6624)

### CODE REVIEW

File: layerInfo.json [hunk 1]
```diff
@@ -151,6 +151,17 @@
 					"@fluid-tools/webpack-fluid-loader"
 				]
 			},
+			"ServiceClients": {
+				"dirs": ["packages/service-clients/"],
+				"deps": [
+					"Container-Definitions",
```
Comment (sonalideshpandemsft) [inline layerInfo.json:157]: Combining the service clients into a single layer

File: layerInfo.json [hunk 2]
```diff
@@ -151,6 +151,17 @@
...
+					"Driver",
+					"Framework",
+					"Runtime",
+					"Routerlicious-Driver",
+					"Server-Shared-Utils"
+				]
+			},
 			"AzureClient": {
 				"dirs": ["azure/packages/"],
 				"deps": [
```
Comment (sonalideshpandemsft) [inline layerInfo.json:167]: not sure if this should be removed. `azure` directory has other packages such as azure-local-service, azure-service-utils and e2e test. 

I'm planning to move azure e2e test to service-clients directory in next follow-up PR.

File: layerInfo.json [hunk 3]
```diff
@@ -99,7 +99,7 @@
 				],
 				"deps": ["Framework-Utils", "Loader", "Runtime", "Client-Utils"]
 			},
-			"ServiceClients": {
+			"TinyliciousClient": {
```
Comment (ChumpChief) [inline layerInfo.json:102]: How challenging is it to move TinyliciousClient into the service-clients directory too?  Not needed for this PR, but would take it if it's trivial.

File: layerInfo.json (hunk 2 above)
Comment (ChumpChief) [reply layerInfo.json:167]: Definitely fine to defer for now, but longer term I could imagine getting rid of the azure directory entirely

**Review by ChumpChief — APPROVED** [review]

File: layerInfo.json (hunk 3 above)
Comment (sonalideshpandemsft) [reply layerInfo.json:102]: shouldn't be difficult I assume. I can have follow-up for this one too

File: layerInfo.json (hunk 2 above)
Comment (sonalideshpandemsft) [reply layerInfo.json:167]: will azure-local-service and azure-service-utils live in service-clients?

File: layerInfo.json (hunk 3 above)
Comment (sonalideshpandemsft) [reply layerInfo.json:102]: https://github.com/microsoft/FluidFramework/pull/19667

**Pull request merged**